android/java/org/chromium/chrome/browser/playlist/hls_content/HlsServiceImpl.java

```diff
@@ -71,136 +69,105 @@ private void startHlsContentFromQueue() {
-                                                                            mPlaylistService
-                                                                                    .updateItemHlsMediaFilePath(
-                                                                                            playlistItem
-                                                                                                    .id,
-                                                                                            mediaPath,
-                                                                                            updatedFileSize);
-                                                                            addNewPlaylistItemModel(
-                                                                                    playlistItem
-                                                                                            .id);
-                                                                            removeContentAndStartNextDownload(
-                                                                                    playlistItem
-                                                                                            .id);
-                                                                        });
-                                                            }
+
+                                mPlaylistService.getPlaylistItem(
+                                        playlistItemId,
+                                        playlistItem -> {
+                                            if (playlistItem == null) {
+                                                PostTask.postTask(
+                                                        TaskTraits.USER_VISIBLE_MAY_BLOCK,
+                                                        () -> {
+                                                            removeContentAndStartNextDownload(
+                                                                    playlistItemId);
                                                         });
                                             }
+                                            currentDownloadingPlaylistItemId = playlistItemId;
+                                            getManifestFile(playlistItem);
                                         });
                             });
                 });
     }
 
+    private void getManifestFile(PlaylistItem playlistItem) {
+        HlsUtils.getManifestFile(
+                mContext,
+                mPlaylistService,
+                playlistItem,
+                new HlsUtils.HlsManifestDelegate() {
+                    @Override
+                    public void onHlsManifestCompleted(Queue<Segment> segmentsQueue) {
+                        String hlsMediaFilePath = HlsUtils.getHlsMediaFilePath(playlistItem);
+                        HlsUtils.deleteFileIfExist(hlsMediaFilePath);
```

Comment:
does that callback get called on bg thread?

Reply:
@SergeyZhukovsky it's happening inside `PostTask` call here : https://github.com/brave/brave-core/blob/e17e343ab5048fcf01c3029e8c0c7e05dfd60b2b/android/java/org/chromium/chrome/browser/playlist/hls_content/HlsServiceImpl.java#L69

Reply:
I see it's called from `HlsUtils.java` `hlsManifestDelegate.onHlsManifestCompleted(segmentsQueue);` and it comes from mojo, all mojo calls I believe executed on UI thread, could you please double check it's UI thread and do IO operations in a PostTask?